src/main/java/io/luna/game/model/World.java

```diff
@@ -149,7 +148,7 @@ public void schedule(Task task) {
      */
     public void queueLogin(Player player) {
         if (player.getState() == EntityState.NEW && !logins.contains(player)) {
-            logins.add(player);
+            logins.offer(player);
```

Comment:
Add and offer are the same thing for ConcurrentLinkedQueue.

Reply:
I assume it doesn't matter in this case since `add` won't ever throw an exception. I've always preferred `offer`, but I'll keep it as `add`.